src/server/protocol.ts

```diff
@@ -30,8 +30,6 @@ export const enum CommandTypes {
     GetSpanOfEnclosingComment = "getSpanOfEnclosingComment",
     Change = "change",
     Close = "close",
-    /** @deprecated Prefer CompletionInfo -- see comment on CompletionsResponse */
-    Completions = "completions",

```

Comment:
This scared me but, [doesn't seem like anyone's actually using it](https://sourcegraph.com/search?q=context:global+Command%28%3F:Names%7CTypes%29%5C.Completions+-repo:.*/Microsoft+-repo:.*/reflec-ts+-repo:.*/tstolua+-repo:.*/typescript-plus+-repo:ts-plus/.*+-file:%5C.d%5C.ts%24+lang:typescript+-repo:.*/UsefulResourceRepo+-repo:.*/CTS+-repo:.*/EmojiScript&patternType=regexp&sm=1&groupBy=repo).

Reply:
I thought of that immediately after closing the tab, I have no idea how to check for that one...

Searching for `executeCommand("completions"`, [it seems not](https://sourcegraph.com/search?q=context:global+executeCommand%5C%28%22completions%22+-repo:.*/Microsoft+-repo:.*/reflec-ts+-repo:.*/tstolua+-repo:.*/typescript-plus+-repo:ts-plus/.*+-file:%5C.d%5C.ts%24+lang:javascript+-repo:.*/UsefulResourceRepo+-repo:.*/CTS+-repo:.*/EmojiScript&patternType=regexp&sm=1&groupBy=repo).

Reply:
This has been deprecated since https://github.com/microsoft/TypeScript/pull/25080

Reply:
I think this change is going to break too many editors.

https://github.com/search?q=tsserver+command+%5C%22completions%5C%22+OR+%5C%27completions%5C%27+NOT+language%3AMarkdown+NOT+CommandTypes&type=code&p=1

https://github.com/MarcWeber/editor-cells/blob/3cb70ffd5e09681b9159bc7527110281ff8660e4/py3/site-packages/cells/asyncio/tsserver.py#L168

https://github.com/angelozerr/typescript.java/blob/8a6759ca825c2b0ae14cb8e09e5ba1778aae24d3/core/ts.core/src/ts/client/CommandNames.java#L25

https://github.com/Hless/atom-typescript/blob/8f67a7770c030b315893fa206bbe8d8d99483a74/lib/client/client.ts#L16

https://github.com/kiyanwang/nvim-typescript/blob/f33d0bc77e475d28bab51c24c7c75142bc3d3013/rplugin/python3/nvim-typescript/client.py#L341

https://github.com/asankah/oni/blob/f139caba34f0c6ef0f85e1305eaf79636ac2a0d1/vim/core/oni-plugin-typescript/src/TypeScriptServerHost.ts#L133-L140

https://github.com/Wilfred/.emacs.d/blob/b0f88ccbd8f41e642b5fad72acf09beb75365d69/elpa/tide-20210930.356/tide.el#L1693-L1695

Reply:
That's definitely a while ago 😅

Reply:
Wow :(

Reply:
Actually, looking again:

* Atom-TypeScript seems updatedish: https://github.com/TypeStrong/atom-typescript/blob/master/lib/client/commandArgsResponseMap.d.ts#L8-L9
* nvim-typescript is deprecated entirely, but latest version is fixed: https://github.com/mhartington/nvim-typescript/blob/44f3db3331632003ed95d963b0753ef94abc8b90/rplugin/node/nvim_typescript/src/client.ts#L20
* oni is deprecated entirely, and oni2 seems to just reuse some VS Code integration? https://github.com/onivim/oni2

* tide actually does need to be updated: https://github.com/ananthakumaran/tide/blob/4cf6a0d89da7f946565a425a632ee2410a40c7da/tide.el#L1702-L1717
* typescript.java actually does need to be updated https://github.com/angelozerr/typescript.java/blob/8a6759ca825c2b0ae14cb8e09e5ba1778aae24d3/core/ts.core/src/ts/client/CommandNames.java#L25
* editor-cells actually does need to be updated: https://github.com/MarcWeber/editor-cells/blob/3cb70ffd5e09681b9159bc7527110281ff8660e4/py3/site-packages/cells/asyncio/tsserver.py#L168
* The Sublime Plugin needs to be updated: https://github.com/microsoft/TypeScript-Sublime-Plugin/blob/fb61dcd11cd3f1807a4bc454148cb59b199e8f3b/typescript/libs/service_proxy.py#L73-L80

Reply:
But does this mean we will never remove the long deprecated server/services API/commands ?

Reply:
Eg Occurrences is another such command

Reply:
I just think that if we want to remove these, we should be willing to send PRs to fix these plugins up.

Reply:
I'd say we can merge this if we send PRs by Monday to the above packages.